### PR TITLE
WW-316 updated copy of to-do list

### PR DESCRIPTION
**Pull request by cormorando**
https://wikia-inc.atlassian.net/browse/WW-316

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/5fe78f16e98d6e1d724b7eb017a9fd145e8c7774/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MINOR [CommunityPageSpecialController.class.php#L42](https://github.com/Wikia/app/blob/5fe78f16e98d6e1d724b7eb017a9fd145e8c7774/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L42): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CommunityPageSpecialController.class.php#L293](https://github.com/Wikia/app/blob/5fe78f16e98d6e1d724b7eb017a9fd145e8c7774/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L293): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/CommunityPage/styles/components/_todo-list-module.scss [hunk 1]
```diff
@@ -31,22 +31,18 @@ $edit-pen-url: '/extensions/wikia/CommunityPage/images/edit-pen-gray.svg';
...
 		padding: 0 10px;
 	}
 
 	.community-page-todo-list-module-zerostate {
 		border-style: solid;
 		border-left: thick solid $color-buttons;
+		clear: both;
 		font-weight: bold;
 		margin-top: 10px;
 		padding: 5px 10px;
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/styles/components/_todo-list-module.scss:49]: the styles of this two classes are almost the same, could we avoid this duplication?

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/cef8d82868c6b1624ef4b9d15258247a07927495/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MINOR [CommunityPageSpecialController.class.php#L42](https://github.com/Wikia/app/blob/cef8d82868c6b1624ef4b9d15258247a07927495/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L42): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CommunityPageSpecialController.class.php#L293](https://github.com/Wikia/app/blob/cef8d82868c6b1624ef4b9d15258247a07927495/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L293): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/CommunityPage/CommunityPage.i18n.php [hunk 2]
```diff
@@ -217,7 +215,7 @@
 	'communitypage-entry-button' => 'Leg los',
 	'communitypage-tasks-title' => 'Aufgaben',
 	'communitypage-entry-tasks' => 'Hilf mit bei der Bearbeitung dieser Community',
-	'communitypage-tasks-header-welcome' => 'Hilf uns dabei, {{SITENAME}} zu erweitern!',
+	'communitypage-tasks-header-welcome' => 'Herzlich Willkommen $1 Fan!',
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/CommunityPage.i18n.php]: This change will break existing functionality and will show `Herzlich Willkommen Fan!`

File: extensions/wikia/CommunityPage/CommunityPage.i18n.php [hunk 3]
```diff
@@ -251,12 +249,12 @@
...
-	'communitypage-popularpages-title' => 'Beliebte Seiten',
-	'communitypage-popularpages-edittext' => 'Lesen und verbessern',
+	'communitypage-popularpages-title' => 'Diese Seiten müssten noch geprüft werden',
+	'communitypage-popularpages-edittext' => 'Lies und korrigiere die Seiten',
 	'communitypage-uncategorizedpages-title' => 'Auf diesen Seiten fehlen noch Kategorien',
 	'communitypage-uncategorizedpages-edittext' => 'Kategorien hinzufügen',
-	'communitypage-wantedpages-title' => 'Gewünschte Seiten',
-	'communitypage-wantedpages-description' => 'Über die roten Links gelangst du zu diesen Seiten, aber die Seiten existieren noch nicht! Erstelle den Artikel oder richte eine Umleitung zum richtigen Artikel ein und erleichtere es damit deinen Lesern, sich in deinem Wiki zurechtzufinden.',
+	'communitypage-wantedpages-title' => 'Wanted pages',
+	'communitypage-wantedpages-description' => "There are redlinks pointing to these pages, but the pages don't exist! Create the article or make it a redirect to the correct article to help readers navigate your wiki.",
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/CommunityPage.i18n.php]: hmmm I'm not sure we want this change.

File: extensions/wikia/CommunityPage/CommunityPage.i18n.php [hunk 4]
```diff
@@ -265,8 +263,8 @@
 	'communitypage-deadendpages-edittext' => 'Links hinzufügen',
 	'communitypage-view-more' => 'Mehr anzeigen',
 	'communitypage-policy-module-link-page-name' => 'Hilfe:Community-Richtlinien',
-	'communitypage-recent-activity-header' => 'Letzte Aktivitäten im Wiki',
-	'communitypage-help-module-title' => 'Hilfe',
+	'communitypage-recent-activity-header' => 'Recent wiki activity',
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/CommunityPage.i18n.php]: I'm not sure we want this change.

File: extensions/wikia/CommunityPage/CommunityPage.i18n.php [hunk 5]
```diff
@@ -321,13 +318,28 @@
...
+	'communitypage-cards-other' => 'Andere',
+	'communitypage-moderators' => 'Die Moderatoren kennenlernen',
 );
 
 $messages['es'] = array(
 	'communitypage-title' => 'Comunidad',
 	'communitypage-header-welcome' => '¡Bienvenido a la comunidad de {{SITENAME}}!',
 	'communitypage-header-join-button' => 'Participa ahora',
-	'communitypage-admin-welcome-message' => 'Comparte tu conocimiento editando páginas, subiendo imágenes y añadiendo nuevo contenido.',
+	'communitypage-admin-welcome-message' => 'Share your knowledge by editing pages, uploading images and adding new content.',
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/CommunityPage.i18n.php]: hmmm I'm not sure we want this change.

File: extensions/wikia/CommunityPage/CommunityPage.i18n.php [hunk 6]
```diff
@@ -443,13 +454,28 @@
...
+	'communitypage-cards-other' => 'Otro',
+	'communitypage-moderators' => 'Conoce a los moderadores',
 );
 
 $messages['fr'] = array(
 	'communitypage-title' => 'Communauté',
 	'communitypage-header-welcome' => 'Bienvenue dans la communauté {{SITENAME}} !',
 	'communitypage-header-join-button' => 'Rejoindre',
-	'communitypage-admin-welcome-message' => 'Partagez vos connaissances en modifiant des pages, en important des images et en ajoutant du contenu.',
+	'communitypage-admin-welcome-message' => 'Share your knowledge by editing pages, uploading images and adding new content.',
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/CommunityPage.i18n.php]: hmmm I'm not sure we want this change.

**Review by dianafa — COMMENTED** [review]
- changes for non-en languages from `{{SITENAME}}` to `$1` will break existing text
- many of strings is not ready yet

**Review by idradm — CHANGES_REQUESTED** [review]
To make it safe to merge, I'd request to revert all changes that are not connected with the sprint 10 translations.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/f8baa2dbcbfa525d080821f8c331cb79905cb0a4/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MINOR [CommunityPageSpecialController.class.php#L42](https://github.com/Wikia/app/blob/f8baa2dbcbfa525d080821f8c331cb79905cb0a4/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L42): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CommunityPageSpecialController.class.php#L293](https://github.com/Wikia/app/blob/f8baa2dbcbfa525d080821f8c331cb79905cb0a4/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L293): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/129a978ea3fca9639141076c3ebf7610a687acab/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MINOR [CommunityPageSpecialController.class.php#L42](https://github.com/Wikia/app/blob/129a978ea3fca9639141076c3ebf7610a687acab/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L42): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
1. MINOR [CommunityPageSpecialController.class.php#L293](https://github.com/Wikia/app/blob/129a978ea3fca9639141076c3ebf7610a687acab/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L293): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [CommunityPageSpecialController.class.php#L3](https://github.com/Wikia/app/blob/b4cd085ff29f49f2aa3751f7ed0cdf56b0e0c2e9/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L3): Class "CommunityPageSpecialController" has 22 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
2. MINOR [CommunityPageSpecialController.class.php#L42](https://github.com/Wikia/app/blob/b4cd085ff29f49f2aa3751f7ed0cdf56b0e0c2e9/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L42): Define a constant instead of duplicating this literal "CommunityPageSpecialController" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
3. MINOR [CommunityPageSpecialController.class.php#L293](https://github.com/Wikia/app/blob/b4cd085ff29f49f2aa3751f7ed0cdf56b0e0c2e9/extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php#L293): Define a constant instead of duplicating this literal "timeAgo" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

File: extensions/wikia/CommunityPage/styles/components/_todo-list-module.scss [hunk 7]
```diff
@@ -31,24 +31,21 @@ $edit-pen-url: '/extensions/wikia/CommunityPage/images/edit-pen-gray.svg';
 		width: $edit-icon-size;
 	}
 
-	.community-page-todo-list-module-description {
+	%community-page-todo-list-module-placeholder {
+		border-style: solid;
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/styles/components/_todo-list-module.scss:35]: do we need this line? `solid` is below

File: extensions/wikia/CommunityPage/styles/components/_todo-list-module.scss [hunk 8]
```diff
@@ -31,24 +31,21 @@ $edit-pen-url: '/extensions/wikia/CommunityPage/images/edit-pen-gray.svg';
...
 		padding: 0 10px;
 	}
 
 	.community-page-todo-list-module-zerostate {
-		border-style: solid;
-		border-left: thick solid $color-buttons;
+		@extend %community-page-todo-list-module-placeholder;
 		font-weight: bold;
-		margin-top: 10px;
 		padding: 5px 10px;
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/styles/components/_todo-list-module.scss:49]: some spacing values have 5 or 10px - could we move it to const. ie. `$padding-base: 5px;` or `$padding-small: 5px; $padding-big: 10px;` ?

File: extensions/wikia/CommunityPage/CommunityPage.i18n.php [hunk 9]
```diff
@@ -321,6 +318,21 @@
 	'communitypage-deadendpages-hightitle' => 'Auf diesen Seiten bitte noch Links einfügen',
 	'communitypage-deadendpages-mediumdescription' => 'Bitte auf diesen Seiten noch Links einfügen. Gut verlinkte Artikel können deinen Lesern noch mehr Informationen zu dem Thema vermitteln und außerdem das Suchmaschinenranking verbessern.',
 	'communitypage-deadendpages-highdescription' => 'Füge Links wichtigen Namen oder Worten hinzu, die in den ersten Sätzen der Seite genannt werden.',
+	'communitypage-subheader-welcome' => 'Lass uns gemeinsam diese Community aufbauen.',
+	'communitypage-admins-welcome-text' => 'Ask us anything. We’d love to hear what brought you to {{SITENAME}}',
+	'communitypage-view-all' => 'Alle anzeigen',
+	'communitypage-help-policy' => 'Community-Regeln',
+	'communitypage-todo-module-description' => 'In diesem Bereich können Admins Seiten vorschlagen, an denen gearbeitet werden sollte und somit neue Autoren bei ihrem Start unterstützen. (Admins: denkt bitte daran, detailliert zu beschreiben, was bearbeitet werden sollte).',
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/CommunityPage.i18n.php:325]: we don't need this key anymore - it has just been removed in this PR

File: extensions/wikia/CommunityPage/CommunityPage.i18n.php [hunk 10]
```diff
@@ -443,6 +454,21 @@
 	'communitypage-deadendpages-hightitle' => 'Por favor añade enlaces a estas páginas',
 	'communitypage-deadendpages-mediumdescription' => 'Por favor añade enlaces a estas páginas. Los artículos bien enlazados ayudan a los lectores a descubrir más información acerca de tu tema y a mejorar la clasificación en los motores de búsqueda.',
 	'communitypage-deadendpages-highdescription' => 'Añade enlaces a cualquier nombre importante o palabra que aparezca en las primeras oraciones de estas páginas.',
+	'communitypage-subheader-welcome' => 'Construyamos juntos esta comunidad.',
+	'communitypage-admins-welcome-text' => 'Ask us anything. We’d love to hear what brought you to {{SITENAME}}',
+	'communitypage-view-all' => 'Ver todo',
+	'communitypage-help-policy' => 'Política de la comunidad',
+	'communitypage-todo-module-description' => 'Esta es un área para que los administradores sugieran páginas que necesitan trabajo para ayudar a los nuevos editores a comenzar. (Administradores - recuerden proporcionar detalles sobre lo que necesita hacerse.)',
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/CommunityPage.i18n.php:461]: we don't need this key anymore - it has just been removed in this PR

File: extensions/wikia/CommunityPage/CommunityPage.i18n.php [hunk 11]
```diff
@@ -565,6 +590,21 @@
 	'communitypage-deadendpages-hightitle' => 'Please add links to these pages',
 	'communitypage-deadendpages-mediumdescription' => 'Please add links to these pages. Well-linked articles help readers discover more information about your topic and improve search engine rankings.',
 	'communitypage-deadendpages-highdescription' => 'Add links to any important names or words that appear in the first few sentences on these pages.',
+	'communitypage-subheader-welcome' => 'Développons cette communauté ensemble.',
+	'communitypage-admins-welcome-text' => 'Ask us anything. We’d love to hear what brought you to {{SITENAME}}',
+	'communitypage-view-all' => 'Tout afficher',
+	'communitypage-help-policy' => 'Règles de la communauté',
+	'communitypage-todo-module-description' => "Les administrateurs peuvent ici suggérer des pages à améliorer, sur lesquelles les nouveaux contributeurs peuvent commencer. (Administrateurs, n'oubliez pas de fournir des détails sur ce qu'il faut faire.)",
```
Comment (dianafa) [inline extensions/wikia/CommunityPage/CommunityPage.i18n.php:597]: we don't need this key anymore - it has just been removed in this PR ... (więcej już nie piszę ;P )

**Pull request closed without merging**